File: 2/Python/main.py
n = int(input("Input the size of the array:\n"))

a = []
result = []

print("Input the array:\n")
for i in range(n):
    a.append(int(input()))

for i in range(n):
    if (i == 0 and max(a[i], a[i+1]) == a[i]) or (i == n-1 and max(a[i], a[i-1]) == a[i]) or (max(max(a[i], a[i+1]), a[i-1]) == a[i]):
        result.append(a[i])

print(result)

File: 2/Scala/Main.scala
//Couldnt run without errors:(
object Main {
    def main(args: Array[String]) = {
        println("Input the size of the array:")
        val n = scala.io.StdIn.readInt()

        var a = Array.ofDim[Int](n)
        var result = scala.collection.mutable.ArrayBuffer[Int]()

        println("Input the array:")
        for (i <- 0 to n) {
            a(i) = scala.io.StdIn.readInt()
        }

        for (i <- 0 to n) {
            if ((i == 0 && math.max(a(i), a(i + 1)) == a(i)) || (i == n - 1 && math.max(a(i), a(i - 1)) == a(i)) || (math.max(math.max(a(i), a(i + 1)), a(i - 1)) == a(i))) {
            result.append(a(i))
            }
        }

        println(result.toArray.mkString(", "))
    }    
}
